### DISCUSSION TITLE
How to properly symlink ~/.kenv or scripts in ~/.kenv/scripts for storing scripts in a repository?

### DISCUSSION THREAD

**Discussion by nicksergeant** [johnlindquist/kit#633] [Q&A] [answered]
My goal is to store my ScriptKit scripts in my dotfiles, but I'm having no luck with the following approaches:

### Symlinking individual scripts into `~/.kenv/scripts/<script>`

When I do this, the script simply doesn't appear in Kit.

### Moving the entire `.kenv` directory into my dotfiles repo (as ie., `~/dotfiles/scriptkit`) and symlinking that entire directory back to `~/.kenv`

When I do this, the default scripts all work fine but any scripts in `~/dotfiles/scriptkit/scripts/` throw an error:

![image](https://user-images.githubusercontent.com/27332/144641268-d70bae3c-ca13-4224-b31f-8b2e2a4fa542.png)

### Creating a new kenv and trying either 1 or 2 above with that.

I had first assumed that kenvs were a way to store scripts in a path other than ~/.kenv but it seems like it's more just a grouping of scripts within my ~/.kenv (I think I'm just confused because they're both called `kenv` 😅).

**Comment by nicksergeant** [comment, marked as answer]
Ahh, [I see now](https://github.com/johnlindquist/kit/discussions/577) that the intended thing to do is to simply init a git repo out of the created `~/.kenv` instead of trying to manage individual scripts.

**Reply by johnlindquist** [reply]
Correct, "shared kenvs" live in the `~/.kenv/kenvs` directory.  So you would see paths like:

* `~/.kenv/kenvs/netlify/scripts`
* `~/.kenv/kenvs/github/scripts`

Kit.app scans for scripts like this:
* `~/.kenv/scripts/**`
* `~/.kenv/kenvs/**/scripts/**`

So a kenv can live anywhere on your machine as long as it symlinks back to `~/.kenv/kenvs/whatever` so Kit.app can find its `scripts` directory.

**Comment by johnlindquist** [comment]
## Symlinking .kenv

If you move your .kenv, then symlink, you also need to:

> From your new directory that symlinks to .kenv
```bash
~/.kit/node/bin/npm i ~/.kit
```

## Why?

In .kenv's `package.json`, `"@johnlindquist/kit"` is a relative path that gets broken, so it needs to be re-configured.

**Reply by nicksergeant** [reply]
Awesome, thanks! I'm still learning how you have this setup and it looks like by having individual repos for `.kenv`s it simplifies the process of sharing entire kenvs with other folks and collaborating on those, so I'll stick with that process for now. :)

**Reply by johnlindquist** [reply]
Cool, just note that shared kenv scripts ignore the schedule/watch/background metadata to avoid those scripts running automatically. We may introduce the concept of a `trusted kenv` in the future.
